### DISCUSSION TITLE
Can I use the results of HYP3 to do Phase Unwrapping

### DISCUSSION THREAD

**Discussion by ZGHHGZ** [opera-adt/tophu#11] [General]
Can I use the results of HYP3 to do Phase Unwrapping

**Comment by gmgunter** [comment]
Hi @ZGHHGZ.

I can't speak to the specifics of HyP3, but this is intended to be a general-purpose unwrapping tool so it should be possible. Assuming that the data is stored in a raster format such as GeoTiff, you could use a tool like GDAL or Rasterio to load the interferogram and coherence data to NumPy arrays that you can then pass to the [`multiscale_unwrap()`](https://github.com/opera-adt/tophu/blob/main/src/tophu/multiscale.py#L363) function.

However, please be advised that this is alpha software without even a single release yet. It might be missing some features that you'd expect, and the interface is very unstable at this point.
